ResourceManager.cpp

```diff
@@ -1,42 +1,41 @@
 #include "ResourceManager.h"
 #include "Archives/VolFile.h"
 #include "Archives/ClmFile.h"
-//#include "Streams/StreamReader.h"
+#include "Streams/SeekableStreamReader.h"
 #include "XFile.h"
 
-using namespace std;
 using namespace Archives;
 
-ResourceManager::ResourceManager(const string& archiveDirectory)
+ResourceManager::ResourceManager(const std::string& archiveDirectory)
 {
-	vector<string> volFilenames = XFile::GetFilesFromDirectory(archiveDirectory, ".vol");
+	std::vector<std::string> volFilenames = XFile::GetFilesFromDirectory(archiveDirectory, ".vol");
```

Comment:
This could be a good place to use `auto`. It saves typing, and really, we are unlikely to care if these are returned as a `vector` or as a `list` or as something else, so long as we can iterate over the filenames in a standard way.

Effectively, we are over-specifying by not using `auto` here. Kind of like specifying `FileStreamReader&` when any `StreamReader&` would do.

Reply:
ok